analytics.py

```diff
@@ -0,0 +1,289 @@
+                        shortestDistance = dist
+            #now add the shortest distance of that point before it moves on to a new point
+            shDistL.append(shortestDistance)
+        #print(shDistL)
+        sums = sum(shDistL)
+        mean_d = sums/len(shDistL)
+        print(mean_d)
+    return mean_d
+
+
+def minimum_bounding_rectangle(points):
+    """
+    Given a set of points, compute the minimum bounding rectangle.
+
+    Parameters
+    ----------
+    points : list
+             A list of points in the form (x,y)
+
+    Returns
+    -------
+     : list
+       Corners of the MBR in the form [xmin, ymin, xmax, ymax]
+    """
+    # a minimum bounding rectangle would be on the extremes of x/y
+
+    xmin = math.inf
+    ymin = math.inf
+    xmax = -9999999999
+    ymax = -9999999999
+    for point in points:
+        if point[0] < xmin:
+            xmin = point[0]
+        if point[1] < ymin:
+            ymin = point[1]
+        if point[0] > xmax:
+            xmax = point[0]
+        if point[1] > ymax:
+            ymax = point[1]
+    mbr = [xmin,ymin,xmax,ymax]
+    print("This is the mbr:")
+    print(mbr)
+    return mbr
+
+def mbr_area(mbr):
+    """
+    Compute the area of a minimum bounding rectangle
+    """
+    length = mbr[2] - mbr[0]
+    width  = mbr[3] - mbr[1]
+    area = length*width
+
+    return area
+
+def expected_distance(area, n):
+    """
+    Compute the expected mean distance given
+    some study area.
+
+    This makes lots of assumptions and is not
+    necessarily how you would want to compute
+    this.  This is just an example of the full
+    analysis pipe, e.g. compute the mean distance
+    and the expected mean distance.
+
+    Parameters
+    ----------
+    area : float
+           The area of the study area
+
+    n : int
+        The number of points
+    """
+
+    expected = 0.5 * (math.sqrt(area/n))
+    return expected
+
+
+def permutation_nearest_distance(mark=[],p=99,n=100):
+    """
+    Finds the nearest neighbor distance for p permutations with n
+    random points
+    :param p: permutation number of times you want to try different
+    simulations for monte carlo
+    :param n: random point number
+    :param mark: Passes in a list of marks if the permutation to be found is of Points
+    :return LDist: list of distances, length p
+    """
+ #   if mark == None:
+ #       LDist = []
+ #       for x in range(p): #loop from 0 to p
+ #           #create n random Points
+ #           points = n_random_points(n) # returns [(x,y),(a,b)..]
+ #           #compute mean neighbor distance
+#            mean_d = average_nearest_neighbor_distance(points)
+ #           LDist.append(mean_d)
+
+    LDist = []
+    for x in range(p): #loop from 0 to p
+        #create n random Points
+        points = utils.n_random_Points(n,mark) # returns [(x,y),(a,b)..]
+        print("print the points array: ")
+        print(points)
+        print(type(points))
```

Comment:
Are these print statements mainly for debugging?  If so, checkout the `logging` module.